components/fetch-build-scan-data-cmdline-tool/build.gradle.kts

```diff
@@ -35,28 +33,31 @@ java {
     }
 }
 
-swaggerSources.register("gradleEnterprise") {
-    setInputFile(layout.projectDirectory.file("src/main/openapi/openapi.yaml").asFile)
-    code.apply {
-        language = "java"
-        components = listOf(
-            "models",
-            "modelTests=false",
-            "modelDocs=false",
-            "apis",
-            "apiTests=false",
-            "apiDocs=false",
-            "supportingFiles",
-            "log.level=error"
-        )
-        configFile = layout.projectDirectory.file("src/main/openapi/openapi-generator-config.json").asFile
-    }
 
-    sourceSets {
-        main {
-            java {
-                srcDir(code)
-            }
+openApiGenerate {
+    generatorName.set("java")
+    inputSpec.set("$projectDir/src/main/openapi/openapi.yaml")
+    outputDir.set("$buildDir/generated/gradle_enterprise_api")
+    ignoreFileOverride.set("$projectDir/.openapi-generator-ignore")
+    modelPackage.set("com.gradle.enterprise.api.model")
+    apiPackage.set("com.gradle.enterprise.api")
+    invokerPackage.set("com.gradle.enterprise.api.client")
+    // see https://github.com/OpenAPITools/openapi-generator/blob/master/docs/generators/java.md for a description of each configuration option
+    configOptions.set(mapOf(
+        "library" to "okhttp-gson",
+        "dateLibrary" to "java8",
+        "hideGenerationTimestamp" to "true",
+        "openApiNullable" to "false",
+        "useBeanValidation" to "false",
+        "disallowAdditionalPropertiesIfNotPresent" to "false",
+        "sourceFolder" to ""  // makes IDEs like IntelliJ more reliably interpret the class packages.
+    ))
+}
+
+sourceSets {
+    main {
+        java {
+            srcDir(tasks.findByName("openApiGenerate")!!)
```

Comment:
findByName resolves the task early:
https://docs.gradle.org/current/userguide/task_configuration_avoidance.html#sec:old_vs_new_configuration_api_overview

Reply:
✅ Good catch!  Fixed.